packages/virtual-list/src/vaadin-virtual-list-mixin.js

```diff
@@ -118,13 +120,31 @@ export const VirtualListMixin = (superClass) =>
 
     /** @private */
     __updateElement(el, index) {
+      if (super.__updateElement) {
+        super.__updateElement(el, index);
+      }
+
       if (el.__renderer !== this.renderer) {
         el.__renderer = this.renderer;
         this.__clearRenderTargetContent(el);
       }
 
       if (this.renderer) {
-        this.renderer(el, this, { item: this.items[index], index });
+        const model = {};
+        this.__updateItemModel(model, this.items[index], index);
+
+        this.renderer(el, this, model);
+      }
+    }
+
+    /**
+     * @private
+     */
+    __updateItemModel(model, item, index) {
+      model.item = this.items[index];
+      model.index = index;
+      if (super.__updateItemModel) {
+        super.__updateItemModel(model, item, index);
```

Comment:
You probably meant `...super.__getItemModel ? super.__getItemModel(index) : {}`? I used this approach earlier but it kind of fealt awkward to return an incomplete model `{ selected: this.__isSelected(...) }` from a function named `__getItemModel`

Reply:
> You probably meant ...super.__getItemModel ? super.__getItemModel(index) : {}?

Yes, sorry for the mistake.

> I used this approach earlier but it kind of fealt awkward to return an incomplete model { selected: this.__isSelected(...) } from a function named __getItemModel

I agree. It seems like the current order in which mixins extend each other is a bit confusing. I would expect that there would be a base mixin that defines the default model, with SelectionMixin extending this base mixin to add its own functionality, rather than the other way around. This is the same concern I raised in https://github.com/vaadin/web-components/pull/8318#discussion_r1879854404.

Reply:
Done. `super.updateElement` need to be called at the end of the overridden function to have the element up-to-date by the time the renderer gets called. That's why I also needed to extract `__updateElementRole` as a separate function.